src/index.js

```diff
@@ -195,12 +195,22 @@ class SWPrecacheWebpackPlugin {
 
   writeServiceWorker(serviceWorker, compiler) {
     const {filepath} = this.workerOptions;
-    const {mkdirp, writeFile} = compiler.outputFileSystem;
+    const {outputFileSystem} = compiler;
```

Comment:
This is weird, is it the way the code is being transpiled?

Why does using `outputFileSystem.mkdirp` vs using `mkdirp` matter?

Reply:
It has to do w/ the way that `memory-fs` stores the filesystem data in memory and (implicitly) binds the `fs` functions. If you look at the code:
https://github.com/webpack/memory-fs/blob/master/lib/MemoryFileSystem.js#L299

You can see that it calls `this[fn + "Sync"](path);` where `this` is implicitly bound. Calling `outputFileSystem.mkdirp` binds `this` to the `outputFileSystem` instance/object. But just calling `mkdirp` or `writeFile` doesn't, so `this` ends up being the global context.

I think it might technically be a flaw of `memory-fs` that they don't explicitly bind those functions to the class instance/object as it's being initialized/created, but it's probably a bit of a gray area and a fairly quick fix here.